snowfakery/cci_mapping_files/declaration_parser.py

```diff
@@ -0,0 +1,118 @@
+import typing as T
+from typing_extensions import Literal
+from datetime import date
+from collections import defaultdict
+
+from pydantic import BaseModel, validator, Extra
+
+
+class MergeRules:
+    @staticmethod
+    def use_highest_priority(
+        new_decl: T.Optional["AtomicDecl"], existing_decl: T.Optional["AtomicDecl"]
+    ):
+        if existing_decl:
+            return max(existing_decl, new_decl, key=lambda decl: decl.priority)
+        return new_decl
+
+    @staticmethod
+    def append(new_decl: "AtomicDecl", existing_decl: T.Optional["AtomicDecl"]):
+        if existing_decl:
+            existing_decl.value.append(new_decl.value)
+            return existing_decl
+        else:
+            d = new_decl
+            # start to build a list-based declaration
+            return AtomicDecl(d.sf_object, d.key, [d.value], d.priority, d.merge_rule)
+
+
+class AtomicDecl(T.NamedTuple):
```

Comment:
You can move this above `MergeRules` and add a doc string :)

Reply:
Done and done.